### PR TITLE
Add ESP8266 tutorial instructions on entering programming mode

### PR DESCRIPTION
**Pull request by ARF1**
This PR adds to the ESP8266 tutorial instructions on how to enter programming programming mode.

### CODE REVIEW

File: docs/esp8266/tutorial/intro.rst [hunk 1]
```diff
@@ -68,6 +68,10 @@ the DTR and RTS pins wired in a special way then deploying the firmware should
 be easy as all steps can be done automatically.  Boards that have such features
 include the Adafruit Feather HUZZAH and NodeMCU boards.
 
+If you do not have such a board, you need keep GPIO0 pulled to ground and reset 
+the device by pulling REST to ground and releasing it again to enter
```
Comment (pfalcon) [inline docs/esp8266/tutorial/intro.rst]: What is:

> REST

?

**Comment by pfalcon** [comment]
So, there's a usual problem: it's definitely better to provide more detailed documentation. But choosing between providing confusing, incomplete, misguiding documentation vs just referring users to external sources of documentation, the latter choice is much better.

In this case: incorrect information is provided on entering a programming mode (well, *incomplete*, but the outcome for a user is the same - frustration, so anyone assessing the impact of providing incomplete documentation should treat it conservatively as incorrect); (apparently) confusing abbreviation is used for "reset". I can't imagine that someone calling _reset_ a "beach", "sun", or "rest" in their good faith (that's why I write "apparently", I can't believe this happens, maybe I'm missing something). But if it happens, I can't think of other underlying motives for doing so, except to cause confusion. And again, confusion is a precursor of frustration.

File: docs/esp8266/tutorial/intro.rst (hunk 1 above)
Comment (mcauser) [inline docs/esp8266/tutorial/intro.rst]: Reset

**Comment by dpgeorge** [comment]
I think this is a useful addition to the docs.

Merged in 2bf1beef5ce2edde80ba7a24dde781fe1b1692a4 with a minor change to change "REST" to "the reset pin".

Thanks for the contribution!

**Pull request closed without merging**